### ISSUE TITLE
Extracting columns that aren't orderable into a different tables results in incorrect data 

### ISSUE THREAD

**Issue by silentninja** [centerofci/mathesar#1490]
## Description
Extracting columns that aren't orderable (e.g., JSON) into a different table using the `move_column` or `split_table` feature results in incorrect data.

## Expected behavior
We should be able to move columns that aren't orderable or give a correct error response if it is not possible.

**Comment by silentninja** [comment]
#1436 needs to be merged for this bug to be valid, marking it as draft until then

**Comment by dmos62** [comment]
For context, we've a predicate function (currently it's private) that tells you if a column is orderable: https://github.com/centerofci/mathesar/blob/73458a90fce21975e87761af32e2a38b59b274e6/db/records/operations/sort.py#L68

**Comment by Reireirei0** [comment]
I'd like to work on this issue, can you assign it to me?

**Comment by rajatvijay** [comment]
@Reireirei0 sure please go ahead. 

Since this will be your first contribution please read our contributing guide here: https://wiki.mathesar.org/en/community/contributing

**Comment by Reireirei0** [comment]
@dmos62 Thanks for your explanation~

**Comment by thesujai** [comment]
@dmos62 @silentninja 

check my modification, is this the desired solution?

def _is_col_orderable(col):
   
    data_type = col.type
    non_orderable_type = ['Binary', 'LargeBinary', 'PickleType', 'ARRAY', 'JSON', 'JSONB']
    if str(data_type) in non_orderable_type:
        raise ValueError("Column is not orderable.")
    return True

**Comment by 7ashraf** [comment]
Can you assign me this issue please?

**Comment by rajatvijay** [comment]
@7ashraf no. See https://github.com/centerofci/mathesar/issues/1490#issuecomment-1481147668

**Comment by dmos62** [comment]
@thesujai you'll want to provide tests that make sure the problem is fixed. Please post a PR before asking for review.

**Comment by ShailRT** [comment]
Where are move_column or split_table function, I want to add try and except logic to this (_is_col_orderable) function.

**Comment by shivansii** [comment]
hey! can I still work on this if the pr is not merged yet?

**Comment by dmos62** [comment]
@shivansii yes, go ahead.

**Comment by esing1201** [comment]
@dmos62 I am happy to help with this issue. Can I work on it?

**Comment by dmos62** [comment]
@esing1201 yes.

**Comment by raisachatterjee** [comment]
Hi, 
I would like to start working on this issue.

**Comment by rai-ankur** [comment]
@seancolsen @seancolsen could you please assign it to me?

**Comment by seancolsen** [comment]
@rai-ankur You're welcome to work on this and submit a PR, but in general we don't assign tickets to new contributors, as noted in our [Contributor Guide](https://github.com/centerofci/mathesar/blob/develop/CONTRIBUTING.md). Once we've already merged a PR of yours, then we'll be happy to assign tickets to you after that. The reason for this policy is that in the past it has been a maintenance burden for us to manage ticket assignments while having an overwhelming volume of people fail to submit their first PR when initially expressing intent to contribute.

**Comment by mathemancer** [comment]
I've added the `db-layer` label since the `split_tables` function is implemented purely in SQL now.

**Comment by kgodey** [comment]
Closing this, we will track column moving improvements and fixes in our roadmap (to be published soon).
